pulser/register/register_layout.py

```diff
@@ -146,24 +178,42 @@ def draw(
             This representation is preferred over drawing the full Rydberg
             radius because it helps in seeing the interactions between atoms.
         """
+        coords = self.coords
         self._draw_checks(
             self.number_of_traps,
             blockade_radius=blockade_radius,
             draw_graph=draw_graph,
             draw_half_radius=draw_half_radius,
         )
         fig, ax = self._initialize_fig_axes(
-            self.coords,
+            coords,
             blockade_radius=blockade_radius,
             draw_half_radius=draw_half_radius,
         )
         self._draw_2D(
             ax,
-            self.coords,
+            coords,
             list(range(self.number_of_traps)),
             blockade_radius=blockade_radius,
             draw_graph=draw_graph,
             draw_half_radius=draw_half_radius,
             are_traps=True,
         )
         plt.show()
+
+    def _safe_hash(self) -> bytes:
+        # Include dimensionality because the array is flattened with tobytes()
+        hash = sha256(bytes(self.dimensionality))
+        hash.update(self.coords.tobytes())
+        return hash.digest()
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, RegisterLayout):
+            return False
+        return self._safe_hash() == other._safe_hash()
+
+    def __hash__(self) -> int:
+        return hash(self._safe_hash())
```

Comment:
Unfortunately no, because `__hash__` returns an `int` and `self._safe_hash()` is in `bytes`. I considered just turning the `bytes` to `int` but since the `hash` in Python has the property of not being the same on different runs, I though it would be best to keep it that way.